src/main/kotlin/org/rust/lang/core/resolve/scope/RustResolveScope.kt

```diff
@@ -1,9 +1,256 @@
 package org.rust.lang.core.resolve.scope
 
-import org.rust.lang.core.psi.RustCompositeElement
-import org.rust.lang.core.psi.RustNamedElement
+import com.intellij.openapi.module.Module
+import com.intellij.psi.PsiElement
+import com.intellij.psi.PsiFile
+import com.intellij.psi.util.PsiTreeUtil
+import com.intellij.psi.util.PsiUtilCore
+import org.rust.cargo.util.AutoInjectedCrates
+import org.rust.cargo.util.cargoProject
+import org.rust.cargo.util.getPsiFor
+import org.rust.cargo.util.preludeModule
+import org.rust.lang.core.psi.*
+import org.rust.lang.core.psi.impl.RustFile
+import org.rust.lang.core.psi.impl.mixin.isStarImport
+import org.rust.lang.core.psi.impl.rustMod
+import org.rust.lang.core.psi.util.module
 
 interface RustResolveScope : RustCompositeElement {
-    val declarations: Collection<RustNamedElement>
+    data class Context(
```

Comment:
> And I think that the entry is the right place to grow things like visibility and priority, which we will need both in the completion and in the resolve.
No, it isn't, since it can be directly derived from PSI.